app/src/main/java/com/cp/tasky/auth/shared/data/repository/AuthRepositoryImpl.kt

```diff
@@ -60,4 +62,22 @@ class AuthRepositoryImpl(
             Result.Error(error = DataError.Remote.UNKNOWN)
         }
     }
+
+    override suspend fun registerUser(userRegisterBody: UserRegisterBody) { //TODO: Handle error cases : Result<Nothing, Error>
+        try {
+            withContext(Dispatchers.IO) {
```

Comment:
That's not necessary, the responsibility to switch the dispatcher lies at the last suspending function that executes the blocking (but not suspending) call. We call that main safety. So in this case, somewhere deep down there will be a blocking function your HTTP lib executes. This function has to be wrapped around withContext, so it's safe to be called from any thread. You can be sure that all the big libraries consider this.